### PR TITLE
python2Packages.pycairo: init at 1.18.2

### PR DESCRIPTION
**Pull request by dotlambda**
###### Motivation for this change
fixes https://github.com/NixOS/nixpkgs/issues/119174

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

We should also try building Gtk2 with Python 3.

### CODE REVIEW

File: pkgs/development/python-modules/pycairo/1.18.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
+{ lib
+, fetchFromGitHub
+, meson
+, ninja
+, buildPythonPackage
+, pytestCheckHook
+, pkg-config
+, cairo
+, isPy3k
+}:
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pycairo/1.18.nix:10]:
```suggestion
, isPy3k
, python
}:
```

File: pkgs/development/python-modules/pycairo/1.18.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  buildInputs = [
+    cairo
+  ];
+
+  checkInputs = [
+    pytestCheckHook
+  ];
+
+  mesonFlags = [
+    "-Dpython=${if isPy3k then "python3" else "python"}"
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pycairo/1.18.nix:40]:
```suggestion
    "-Dpython=${python.interpreter}"
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Does this need to target staging?

**Comment by dotlambda** [comment]
> Does this need to target staging?

It doesn't make sense not to cause `python3Packages.pycairo` is on 1.18.2 in `staging-next`.

File: pkgs/development/python-modules/pycairo/default.nix [hunk 3]
```diff
@@ -37,7 +40,7 @@ buildPythonPackage rec {
   ];
 
   mesonFlags = [
-    "-Dpython=${if isPy3k then "python3" else "python"}"
+    "-Dpython=${python.interpreter}"
   ];
 
   meta = with lib; {
```
Comment (davidak) [inline pkgs/development/python-modules/pycairo/default.nix:46]: remove python 3 from description

**Comment by davidak** [comment]
In the linked issue, gimp needed the python 2 version, so should it be used in this PR or in a different?

I tried to build gimp from this PR, but it also builds LLVM which takes forever, so it stopped it.

File: pkgs/top-level/python2-packages.nix [hunk 4]
```diff
@@ -391,6 +391,10 @@ with self; with super; {
 
   pyblosxom = callPackage ../development/python-modules/pyblosxom { };
 
+  pycairo = callPackage ../development/python-modules/pycairo/1.18.nix {
```
Comment (SuperSandro2000) [inline pkgs/top-level/python2-packages.nix:394]: Isn't the default.nix now unused or do I miss something?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 119185` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>gimp</li>
</ul>
</details>

File: pkgs/development/python-modules/pycairo/1.18.nix [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  checkInputs = [
+    pytestCheckHook
+  ];
+
+  mesonFlags = [
+    "-Dpython=${python.interpreter}"
+  ];
+
+  meta = with lib; {
+    description = "Python 2/3 bindings for cairo";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pycairo/1.18.nix:44]:
```suggestion
    description = "Python 2 bindings for cairo";
```

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/development/python-modules/pycairo/1.18.nix (hunk 5 above)
Comment (dotlambda) [reply pkgs/development/python-modules/pycairo/1.18.nix:44]: That package also works on Python 3.

Comment (davidak) [reply pkgs/development/python-modules/pycairo/1.18.nix:44]: yes, but it is not available for python 3 for end-users of nix?

**Review by davidak — APPROVED** [review]
Looks good

**Pull request merged**